web/js/MapUI/Components/UpdateProjection/UpdateProjection.js

```diff
@@ -0,0 +1,399 @@
+  *
+  * @returns {void}
+  */
+  const clearLayers = function() {
+    const activeLayers = ui.selected
+      .getLayers()
+      .getArray()
+      .slice(0);
+    lodashEach(activeLayers, (mapLayer) => {
+      ui.selected.removeLayer(mapLayer);
+    });
+    ui.cache.clear();
+  };
+
+  /**
+   * @method reloadLayers
+   *
+   * @param {object} map - Openlayers Map obj
+   * @param {Object} granuleOptions (optional: only used for granule layers)
+   *   @param {Boolean} granuleDates - array of granule dates
+   *   @param {Boolean} id - layer id
+   * @returns {void}
+   */
+  async function reloadLayers(granuleOptions) {
+    const map = ui.selected;
+    const { createLayer } = ui;
+
+    if (!config.features.compare || !compare.active) {
+      const compareMapDestroyed = !compare.active && compareMapUi.active;
+      if (compareMapDestroyed) {
+        compareMapUi.destroy();
+      }
+      clearLayers();
+      const defs = getLayers(layerState, { reverse: true });
+      const layerPromises = defs.map((def) => {
+        const options = getGranuleOptions(layerState, def, compare.activeString, granuleOptions);
+        return createLayer(def, options);
+      });
+      const createdLayers = await Promise.all(layerPromises);
+      lodashEach(createdLayers, (l) => { map.addLayer(l); });
+    } else {
+      const stateArray = [['active', 'selected'], ['activeB', 'selectedB']];
+      if (compare && !compare.isCompareA && compare.mode === 'spy') {
+        stateArray.reverse(); // Set Layer order based on active A|B group
+      }
+      clearLayers();
+      const stateArrayGroups = stateArray.map(async (arr) => getCompareLayerGroup(arr, layerState, granuleOptions));
+      const compareLayerGroups = await Promise.all(stateArrayGroups);
+      compareLayerGroups.forEach((layerGroup) => map.addLayer(layerGroup));
+      compareMapUi.create(map, compare.mode);
+    }
+    updateLayerVisibilities();
+  }
+
+  const onStopAnimation = function() {
+    const needsRefresh = activeLayers.some(({ type }) => type === 'granule' || type === 'vector');
+    if (needsRefresh) {
+      // The SELECT_DATE and STOP_ANIMATION actions happen back to back and both
+      // try to modify map layers asynchronously so we need to set a timeout to allow
+      // the updateDate() function to complete before trying to call reloadLayers() here
+      setTimeout(reloadLayers, 100);
+    }
+  };
+
+  /**
+ * When page is resized set for mobile or desktop
+ *
+ * @method onResize
+ * @static
+ *
+ * @returns {void}
+ */
+  function onResize() {
+    const map = ui.selected;
+    if (isMobile) {
+      map.removeControl(map.wv.scaleImperial);
+      map.removeControl(map.wv.scaleMetric);
+    } else {
+      map.addControl(map.wv.scaleImperial);
+      map.addControl(map.wv.scaleMetric);
+    }
+  }
+
+  /**
+ * Show Map
+ *
+ * @method showMap
+ * @static
+ *
+ * @param {object} map - Openlayers Map obj
+ *
+ * @returns {void}
+ */
+  function showMap(map) {
```

Comment:
"map" is already declared on line 42.

Reply:
This map parameter does represent a different variable than the map variable declared in props from `mapStateToProps`.